utilities/backupable/backupable_db.cc

```diff
@@ -178,12 +178,12 @@ class BackupEngineImpl {
  private:
   void DeleteChildren(const std::string& dir,
                       uint32_t file_type_filter = 0) const;
-  Status DeleteBackupNoGC(BackupID backup_id);
+  IOStatus DeleteBackupNoGC(BackupID backup_id);
 
   // Extends the "result" map with pathname->size mappings for the contents of
   // "dir" in "env". Pathnames are prefixed with "dir".
-  Status ReadChildFileCurrentSizes(
-      const std::string& dir, Env* env,
+  IOStatus ReadChildFileCurrentSizes(
+      const std::string& dir, const std::shared_ptr<FileSystem>&,
```

Comment:
This is probably not the time and place, but...

I think `const std::shared_ptr<X>&` rarely makes sense as a parameter type. If the called function should never save a copy of the pointer, use X* to save indirection. If the called function (almost) always saves a copy of the pointer, use std::shared_ptr<X> for simplicity and so that we don't force unnecessary increment+decrement the refcount if the caller doesn't need to keep a copy, e.g. passing `std::make_shared<X>()`

If the called function may or may not save a copy, OK `const std::shared_ptr<FileSystem>&`. At least in my mind, these different types help to implicitly document how the pointer is going to be used.

Reply:
I see. Your statement make senes to me that if saving a copy is not required for a std::shared_ptr<>, then passing const std::shared_ptr<FileSystem>& is ok. However, in this case, some of the functions being called like LineFileReader::Create and IOStatus SequentialFileReader::Create(), takes const std::shared_ptr<FileSystem>&, that's why I'm thinking to take it.

Reply:
Right. I mean to criticize the existing pattern. :) Your PR just made me think of it.